accepted/cranelift-backend-transition.md

```diff
@@ -0,0 +1,230 @@
+   missed, or helping users to select the old backend, if appropriate.
+   
+5. Make several releases with the new backend as default. Keep the old
+   backend alive with CI; it will be "fully supported" during this
+   time, but its use will be deprecated.
+
+6. At some future point, when we are confident that no significant
+   use-cases or dependencies remain, we can remove the old
+   backend. This will be a separate RFC process with its own
+   consensus-gathering.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This transition's rationale largely derives from the rationale for the
+new backend in general: the new design is simpler, more maintainable,
+generally more performant, and has a more trustworthy security
+stance. Given these benefits, if there are no blocking issues, it
+makes sense to switch the default.
+
+Furthermore, there is a continuing cost to maintaining support for the
+old backend. Its presence in the codebase adds complications because
+many data-structures and code paths have to account for both
+cases. When we add features that require backend support, we have to
+evaluate a tricky tradeoff question: do we spend time adding the
+feature to the old backend as well, given that at some future point it
+will be removed? If such work is necessary to support a feature in the
+*default* build today, then extra work is required that will
+eventually no longer be useful.
+
+Because of this cost and the tradeoffs of spending more effort on the
+old backend, it is already stagnating slightly in terms of feature
+support. For example, its SIMD support is not up-to-date with respect
+to the latest Wasm SIMD proposal. As other Wasm proposals advance, it
+will likely fall further behind and time spent updating it will be
+harder and harder to justify.
+
+The major alternative is simply to do nothing: retain the old backend as
+default, and new backend as a non-default option. This is the status-quo and
+has the least risk; individual embedders can always enable the new backend if
+desired.  However, it has the significant downside that it implies ongoing
+support for the old backend; in the steady state, this is twice the maintenance
+and possibly feature-implementation work for one platform (x86-64), and so we
+do not consider this to be a serious option *unless* the new backend shows
+serious defects that prevent switching.
+
+In other words, given the direction that effort is being, and will be, spent,
+and given the effort and maintenance burden of various options in the future,
+it seems to be inevitable that the switch will occur at some point. The
+question to answer is whether the new backend is ready *yet* to be the default,
+or whether we need to wait longer.
+
+# Open questions
+[open-questions]: #open-questions
+
+1. How much compile-time and runtime performance degradation should we
+   accept, if any, before moving forward with this transition?
```

Comment:
In order to help guiding the answer to this question, how about another question: from the performance analysis that we've made on the new backend, it would be interesting to get a rough idea of the speedups we can reach, and with which associated cost? What I mean is, if we think we can get e.g. a 10% speedup **easily** (famous last words, but imagine in e.g. one or two months of work), then it makes it easier to accept a degradation in the short-term.